src/core/validation/validator.ts

```diff
@@ -370,13 +391,34 @@ export class Validator {
         }
       }
     }
-    
+
     // Fallback to filename without extension if not in expected structure
     const fileName = parts[parts.length - 1] ?? '';
     const dotIndex = fileName.lastIndexOf('.');
     return dotIndex > 0 ? fileName.slice(0, dotIndex) : fileName;
   }
 
+  private extractSectionPreview(content: string, sectionNames: string[]): string {
```

Comment:
The preview helper only finds section headers when the line exactly equals "## ADDED Requirements" (same case/spacing). But `parseDeltaSpec` accepts headers case-insensitively and with flexible whitespace, so authors who write ## Added Requirements or add extra spaces never see the “First few lines…” block, even though we already marked the section as present. 

Please normalize the header before comparing (trim, collapse spaces, lowercase) so this finds ## Added Requirements or headers with extra spaces, not just the exact "## ADDED Requirements" string. Currently those variants are accepted by the parser but never show a preview.